code-challenges.test.js

```diff
@@ -15,35 +15,169 @@
+
+
 
 // --------------------2) Create a function that takes an array of numbers and returns an array of the minimum and maximum numbers in that order.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const nums1 = [3, 56, 90, -8, 0, 23, 6]
-// Expected output: [-8, 90]
-const nums2 = [109, 5, 9, 67, 8, 24]
-// Expected output: [5, 109]
+describe("minAndMax", () => {
+    it("takes an array of numbers and returns an array of the minimum and maximum numbers in that order", () => {
+        const nums1 = [3, 56, 90, -8, 0, 23, 6]
+            // Expected output: [-8, 90]
+        const nums2 = [109, 5, 9, 67, 8, 24]
+            // Expected output: [5, 109]
+      expect(minAndMax(nums1)).toEqual([-8, 90])
+      expect(minAndMax(nums2)).toEqual([5, 109])
+    })
+  })
+
+//  ReferenceError: minAndMax is not defined
+
+
+// b) Create the function that makes the test pass/
+
+    // Psuedo code:
+        // declare a function 
+        // set param array 
+        // declare a variable 
+        // use array.max and array.min to find the minimum and maximum values
+        // assign results to (array.min, array.max)
+
+// const minAndMax = (array) => {
+//     let min = Math.min(array).toString;
+//     let max = Math.max(array).toString;
+//     return min.split()
+// }
+
+
+
+// const minAndMax = (array) => {
+//     let results = ((array.min, array.max));
+//      let newArray = [];
+//      newArray.push(results)
+//      return newArrayS
+// }
+//
+
+// const minAndMax = (array) => {
+//     return array.filter ((value) => {
+//         if ((Math.min(array)) <= value) {
+//         return value
+//         } else if ( (Math.max(array)) >= value) {
+//         return value 
+//         }
+//     })
+// }
+
+
+//-------Psuedo Code------//
+    // declare a function called minAndMax
+    // set param (array)
+    // declare a variable called newArray and assign it to a function that returns an array in ascending order
+    // use .filter to iterate through the new array
+    // use condtional statements with Math.min() and Math.max() to determine which values are the minimum and maximum
+        //if Math.min(newArray) === value return value
+        // else if Math.max(newArray) === value return value 
+    // return results 
+
+const minAndMax = (array) => {
+    let newArray = array.sort((a,b) => {return a - b})
+    return newArray.filter (value => {
+        if (Math.min(...newArray) === value) {
+            return value
+        } else if ( Math.max(...newArray) === value) {
+            return value 
+        }
+    })
+}
+
+// Test Suites: 1 passed, 1 total
+
+// Its 10:40 pm, I honestly have no idea what I am doing wrong here. It works when I console.log it but not when I test it. Whyyyyyyy?!
+// LOL false alarm, i figured it out. its 10:45 and been looong day. 
```

Comment:
Glad you figured it out!  Great work.  Love that you showed all your attempts.  You personal comments crack me up.  Thank you!